htmresearch/frameworks/nlp/classify_fingerprint.py

```diff
@@ -131,3 +134,101 @@ def testModel(self, i, seed=42):
     (_, inferenceResult, _, _) = self.classifier.infer(self.sparsifyPattern(
       self.patterns[i]["pattern"]["bitmap"], self.encoder.n))
     return self.getWinningLabels(inferenceResult, seed)
+
+
+  def trainText(self, token, labels, sequenceId=None, reset=0):
+    """
+    Train the model with the given text token, associated labels, and
+    sequence ID. The sequence ID is stored in sampleReference so we know which
+    samples the model has been trained on, and specifically where they
+    appear in the classifier space.
+
+    @param token      (str)  The text token to train on
+    @param labels     (list) A list of one or more integer labels associated
+                             with this token. If the list is empty, the
+                             classifier will not be trained.
+    @param sequenceId (int)  An integer ID associated with this token and its
+                             sequence (document).
+    @param reset      (int)  Should be 0 or 1. If 1, assumes we are at the
+                             beginning of a new sequence.
+    """
+    if self.currentDocument is None:
+      # start of a new document
+      self.currentDocument = [token]
+    else:
+      # accumulate text for this document
+      self.currentDocument.append(token)
+
+    self.currentDocumentId = sequenceId
```

Comment:
You don't need self.currentDocumentId. It is only ever used within the scope of this method.